src/main/java/de/tum/in/www1/artemis/service/exam/StudentExamService.java

```diff
@@ -314,6 +312,40 @@ else if (exercise instanceof ModelingExercise) {
         return studentExams;
     }
 
+    /**
+     * Helper method to return a map for each user to their exercises
+     *
+     * @param studentExams the student exams of the users containing the exercises
+     * @return a map of the User as key, and a list of the users exercises as value
+     */
+    public Map<User, List<Exercise>> getExercisesOfUserMap(Set<StudentExam> studentExams) {
```

Comment:
Consider making this method private or at least package private. It is only a helper method and probably won't be used outside of this class. Usually helper methods are tested through the methods calling them.

Reply:
It is used outside of this class in the test classes, so I think it should stay public